source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1211 @@
+
+   * - `first() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#first()>`__
+     - :manual:`$first </reference/operator/aggregation/first-array-element/>`
+
+   * - `joinStrings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#joinStrings(java.util.function.Function)>`__
+     - :manual:`$concat </reference/operator/aggregation/concat/>`
+
+   * - `last() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#last()>`__
+     - :manual:`$last </reference/operator/aggregation/last-array-element/>`
+
+   * - `map() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#map(java.util.function.Function)>`__
+     - :manual:`$map </reference/operator/aggregation/map/>`
+
+   * - `max() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#max(T)>`__
+     - :manual:`$max </reference/operator/aggregation/max/>`
+
+   * - `maxN() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#maxN(com.mongodb.client.model.mql.MqlInteger)>`__
+     - :manual:`$maxN </reference/operator/aggregation/maxN-array-element/>`
+
+   * - `min() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#min(T)>`__
+     - :manual:`$min </reference/operator/aggregation/min/>`
+
+   * - `minN() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#minN(com.mongodb.client.model.mql.MqlInteger)>`__
+     - :manual:`$map </reference/operator/aggregation/minN-array-element/>`
+
+   * - `multiply() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#multiply(java.util.function.Function)>`__
+     - :manual:`$multiply </reference/operator/aggregation/multiply/>`
+
+   * - `size() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#size()>`__
+     - :manual:`$size </reference/operator/aggregation/size/>`
+
+   * - `slice() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#slice(int,int)>`__
+     - :manual:`$slice </reference/operator/aggregation/slice/>`
+
+   * - `sum() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#sum(java.util.function.Function)>`__
+     - :manual:`$sum </reference/operator/aggregation/sum/>`
+
+   * - `union() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#union(com.mongodb.client.model.mql.MqlArray)>`__
+     - :manual:`$setUnion </reference/operator/aggregation/setUnion/>`
+
+   * - `unionArrays() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#unionArrays(java.util.function.Function)>`__
+     - :manual:`$setUnion </reference/operator/aggregation/setUnion/>`
+
+.. tip::
+
+   When retrieving arrays, you must specify the type of array that is returned
+   if you want to continue working with the return value.
```

Comment:
It might help if this had the example in it, I think it is referring to specifying MqlDocument in `current().<MqlDocument>getArray("showtimes");`? And the "continue working with", maybe "if you need to work with the values of the array as that specific type".

Maybe it is better to put this point together with the example, like "the getArray method is used to ... . Specifying the type using <MqlDocument> ensures that `showtime` is an `MqlDocument`" or something like this.

I am unsure, because the "tip" does highlight this important point which might be skipped if it was buried in a paragraph, but when I read it here, it's not clear what it means to specify the type of the array, or to continue working with the return value (note that you can continue working with the return value even without specifying, since it will just assume that the elements are of the general type MqlValue).

Reply:
I will leave it as a tip but move it down and change the content